File: src/c/assumption.c
#include "./include/assumption.h"

void assumption_sort(void *data, size_t len, size_t stride, int (*compar)(const void *, const void *)) {
  // Assume that the array is sorted.
  (void) stride;
  (void) len;
  (void) data;
  (void) compar;
}

void assumption_quick_sort(void *data, size_t len, size_t stride, int (*compar)(const void *, const void *)) {
  // Assume that the array is sorted.
}

File: src/fortran/assumption.f90
subroutine assumption_sort(arr, compare_function)
  integer, dimension(:) :: arr
  interface
    logical function compare_function(x, y)
      integer, intent(in) :: x, y
    end function compare_function
  end interface

  ! Assume that the array is already sorted.

end subroutine assumption_sort
